python/Ganga/Runtime/bootstrap.py

```diff
@@ -762,14 +762,13 @@ def initEnvironment(opt_rexec):
 
             
             import inspect
+            from os.path import expandvars, expanduser
+
             GangaRootPath = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe()))), '../..'))
             def transform(x):
                 return os.path.normpath(Ganga.Utility.files.expandfilename(os.path.join(GangaRootPath,x)))
 
-            from os.path import expandvars, expanduser
-
-            paths = map(transform, filter(lambda x: expandvars(expanduser(x)), config['RUNTIME_PATH'].split(':')))
-
+            paths = map(transform, filter(None, map(lambda x: expandvars(expanduser(x)), config['RUNTIME_PATH'].split(':'))))
```

Comment:
also (I know you didn't start it.. It may well have even been me ;-)) I think that all this use for map/filter/lambda is quite messy.

could you check if something like:
```python
paths = [os.normpath(os.normcase(os.expandvars(os.expanduser(p.strip())))) 
             for p in config['RUNTIME_PATH'].split(':') if p.strip()]
```

works in place here?

If so don't change your pull request and we can discuss in the meeting (since this must be a common thing to need to do) if it would probably be an idea to update [Ganga.Utility.files.expandfilename](https://github.com/ganga-devs/ganga/blob/develop/python/Ganga/Utility/files.py#L15)

to something like:

```python
from os.path import abspath, expandvars, expanduser, normcase
def iexpandfilenames(filenames, force=False, ignore_empty=True):
    if not isinstance(filenames, Iterable):
        filenames = [filenames]

    for f in filenames:
        sanitised_path = abspath(normcase(expandvars(expanduser(f.strip()))))
        if ignore_empty:
            continue
        if os.path.exists(sanitised_path) or force:
            yield sanitised_path
        else:
            getLogger().debug("Filename: %s doesn't exist using it anyway" % f.strip())
            yield f.strip()

def expandfilenames(filenames, force=False):
    return [f for f in iexpandfilenames(filenames, force)]
```

and then this line would become simply:

```python
paths = [os.path.join(GangaRootPath,p) for p in 
              Ganga.Utility.files.iexpandfilenames(config['RUNTIME_PATH'].split(':'))]
```